Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -12001,6 +12026,313 @@ def pan_os_delete_redistribution_profile_command(args):
     )
 
 
+def build_pbf_xpath(name, pre_post, element_to_change=None):
+    _xpath = f'{XPATH_RULEBASE}{pre_post}/pbf'
+    if name:
+        _xpath = f"{_xpath}/rules/entry[@name='{name}']"
+    if element_to_change:
+        _xpath = f'{_xpath}/{element_to_change}'
+    return _xpath
+
+
+def pan_os_list_pbf_rules(name, pre_post, show_uncommitted):
+
+    if DEVICE_GROUP and not pre_post:  # panorama instances must have the pre_post argument!
+        raise DemistoException(f'The pre_post argument must be provided for panorama instance')
+
+    params = {
+        'type': 'config',
+        'action': 'get' if show_uncommitted else 'show',
+        'key': API_KEY,
+        # rulebase is for firewall instance.
+        'xpath': build_pbf_xpath(name, 'rulebase' if VSYS else pre_post)  # type: ignore[arg-type]
+    }
+
+    return http_request(URL, 'GET', params=params)
+
+
+def parse_pan_os_list_pbf_rules(entries, show_uncommitted):
+    if show_uncommitted:
+        for entry in entries:
+            parse_pan_os_un_committed_data(entry, keys_to_remove=['@admin', '@time', '@dirtyId', '@uuid', '@loc'])
+
+    human_readable, context = [], []
+
+    for entry in entries:
+        human_readable.append(
+            {
+                'Name': extract_objects_info_by_key(entry, '@name'),
```

Comment:
can we extract some of these variables to values to reuse them?

Reply:
will be done